Dockerfile.windows

```diff
@@ -90,27 +101,31 @@ RUN cmake --build . --config Release; `
 
 # TODO: Do something with /work/out/bin/fluent-bit.{exe,dll}
 
+###############################################################################
+# Build Go code in one container to exploit Go build caching
+###############################################################################
+
+FROM base as gobuilder
+
 ###############################################################################
 # Build OT collector
 ###############################################################################
 
-WORKDIR /work/submodules/opentelemetry-operations-collector
+COPY submodules/opentelemetry-operations-collector /work/submodules/opentelemetry-operations-collector
 
-RUN go get github.com/golangci/golangci-lint/cmd/golangci-lint; `
-    go get github.com/google/addlicense; `
-    go get github.com/pavius/impi/cmd/impi;
+WORKDIR /work/submodules/opentelemetry-operations-collector
 
 RUN go build -o bin/google-cloud-metrics-agent_windows_amd64.exe ./cmd/otelopscol; `
     Copy-Item -Path bin/google-cloud-metrics-agent_windows_amd64.exe -Destination /work/out/bin/;
 
-# TODO: Do something with /work/out/bin/google-cloud-metrics-agent_windows_amd64.exe
-
 ###############################################################################
 # Config generator/service installer
 ###############################################################################
 
 WORKDIR /work
 
+COPY . /work
```

Comment:
Might make sense to keep the WORKDIR close to the RUN commands, so I'd move the COPY above the WORKDIR. Ditto below. As I've already approved, please merge after addressing.

Reply:
I prefer the current order because the COPY is also a command that mutates the /work directory.